eras/allegra/impl/src/Cardano/Ledger/Allegra/TxAuxData.hs

```diff
@@ -74,16 +75,18 @@ import NoThunks.Class (NoThunks)
 data AllegraTxAuxDataRaw era = AllegraTxAuxDataRaw
   { atadrMetadata :: !(Map Word64 Metadatum)
   -- ^ Structured transaction metadata
-  , atadrTimelock :: !(StrictSeq (Timelock era))
+  , atadrNative :: !(StrictSeq (NativeScript era))
   -- ^ Pre-images of script hashes found within the TxBody, but which are not
   -- required as witnesses. Examples include:
   -- - Token policy IDs appearing in transaction outputs
   -- - Pool reward account registrations
   }
-  deriving (Generic, Eq)
+  deriving (Generic)
+
+deriving instance Eq (NativeScript era) => Eq (AllegraTxAuxDataRaw era)
 
 class EraTxAuxData era => AllegraEraTxAuxData era where
-  timelockScriptsTxAuxDataL :: Lens' (TxAuxData era) (StrictSeq (Timelock era))
+  nativeScriptsTxAuxDataL :: Lens' (TxAuxData era) (StrictSeq (NativeScript era))
```

Comment:
Instead of removing it, let's help downstream users migrate to the new lens with the help of deprecation:

```suggestion
  nativeScriptsTxAuxDataL :: Lens' (TxAuxData era) (StrictSeq (NativeScript era))
  
  timelockScriptsTxAuxDataL :: Lens' (TxAuxData era) (StrictSeq (Timelock era))
  default timelockScriptsTxAuxDataL :: NativeScript era ~ Timelock era => Lens' (TxAuxData era) (StrictSeq (Timelock era))
  timelockScriptsTxAuxDataL = nativeScriptsTxAuxDataL

{-# DEPRECATED timelockScriptsTxAuxDataL "In favor of `nativeScriptsTxAuxDataL`" #-}
```